climada/util/yearsets.py

```diff
@@ -22,223 +22,227 @@
-            The sampling dictionary containing two arrays:
-                selected_events (array): sampled events (len: total amount of sampled events)
-                events_per_year (array): events per sampled year
-        n_events_per_year : array
+        events_per_year : array
             Number of events per sampled year
     """
+    if not lam:
+        lam = np.sum(imp.frequency)
 
     #sample number of events per year
-    if n_annual_events != 1:
-        events_per_year = np.round(np.random.poisson(lam=n_annual_events,
+    if lam != 1:
+        events_per_year = np.round(np.random.poisson(lam=lam,
                                                      size=n_sampled_years)).astype('int')
     else:
         events_per_year = np.ones(len(n_sampled_years))
 
-    tot_n_events = np.sum(events_per_year)
-
-    selected_events = sample_events(tot_n_events, n_input_events)
-
-    sampling_dict = dict()
-    sampling_dict = {'selected_events': selected_events, 'events_per_year': events_per_year}
 
-    return sampling_dict
+    return events_per_year
 
-def sample_events(tot_n_events, n_input_events):
-    """Sample events (length = tot_n_events) uniformely from an array (n_input_events)
-    without replacement (if tot_n_events > n_input_events the input events are repeated
-                         (tot_n_events/n_input_events) times).
+def sample_events(events_per_year, freqs_orig):
+    """Sample events uniformely from an array (indices_orig) without replacement
+    (if sum(events_per_year) > n_input_events the input events are repeated
+     (tot_n_events/n_input_events) times, by ensuring that the same events doens't
+     occur more than once per sampled year).
 
     Parameters:
-        tot_n_events : int
-            Number of events to be sampled
-        n_input_events : int
-         Number of events contained in given event_impacts object
+        events_per_year : array
+            Number of events per sampled year
+        freqs_orig : array
+            Frequency of each input event
 
     Returns:
-        selected_events : array
-            Uniformaly sampled events (length: tot_n_events)
+        sampling_vect : 2D array
+            The sampling vector specifies how to sample the yimp, it consists of one
+            sub-array per sampled_year, which contains the event_ids of the events used to
+            calculate the annual impacts.
       """
 
-    repetitions = np.ceil(tot_n_events/n_input_events).astype('int')
-    indices = np.tile(np.arange(n_input_events), repetitions)
+    sampling_vect = []
 
-    rng = default_rng()
-    selected_events = rng.choice(indices, size=tot_n_events,
-                                 replace=False).astype('int')
+    indices_orig = np.arange(len(freqs_orig))
 
+    freqs = freqs_orig
+    indices = indices_orig
 
-    return selected_events
+    #if check_doubling true: check that every event doesn't occur more than once per sampled year
+    check_doubling = False
```

Comment:
That should be an input variable or be removed.

Reply:
I implemented this variable so the doubling is only checked when the same event is actually contained twice in the indices variable to reduce the use of the while loop.

Reply:
Maybe could you rephrase the comment just to make it clear that this is not an option to turn off the double counting?